Assets/BowAndArrowImproved/Scripts/Leaderboard/Leaderboard.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+public class Leaderboard : MonoBehaviour
+{
+    private const string JsonLeaderBoardFileName = "JsonLeaderboardHolder.json";
+    private LeaderboardEntries _leaderboardEntries;
+
+    private IDataLoader _leaderboardLoader = new JsonDataLoader();
+    private IDataSaver _leaderboardSaver = new JsonDataSaver();
```

Comment:
that's a good case for a factory class, since `Leaderboard` doesn't really care what loader/saver it gets, it just want a data loader/saver, create class called `DataHandlerFactory` which contains a single `static` method called `get()`, the signature of this function should be something like:
```
public static IDataHandler get()
```
and inside it, create a `JsonDataHandler` instance and return it - then, here, simply use the factory class like so:
```
private IDataHander _dataHandler = DataHandlerFactory.get();
```